app/components/post_list/post_list.js

```diff
@@ -24,37 +24,74 @@ const style = StyleSheet.create({
     }
 });
 
-export default class PostList extends React.Component {
+const LOAD_MORE_POSTS = 'load-more-posts';
+
+export default class PostList extends Component {
     static propTypes = {
-        posts: React.PropTypes.array.isRequired,
-        theme: React.PropTypes.object.isRequired,
-        onPostPress: React.PropTypes.func,
-        renderReplies: React.PropTypes.bool,
-        indicateNewMessages: React.PropTypes.bool,
-        currentUserId: React.PropTypes.string,
-        lastViewedAt: React.PropTypes.number
+        posts: PropTypes.array.isRequired,
+        postsRequests: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        allowLoadMore: PropTypes.bool,
+        loadMore: PropTypes.func,
+        onPostPress: PropTypes.func,
+        renderReplies: PropTypes.bool,
+        indicateNewMessages: PropTypes.bool,
+        currentUserId: PropTypes.string,
+        lastViewedAt: PropTypes.number
     };
 
     constructor(props) {
         super(props);
-        const {posts, indicateNewMessages, currentUserId, lastViewedAt} = this.props;
         this.state = {
+            hasFirstPost: false,
```

Comment:
@hmhealey That was also my first thought; we should track the posts list pagination state in the store. After all, we're doing so in the webapp (with `atTop` & `atBottom` in `post_store.jsx`). But when I followed that path, I encountered an obstacle: we are dispatching `RECEIVED_POSTS` for all of these actions: `getPost`, `getPosts`, `getPostsSince`, `getPostsBefore`, and `getPostsAfter`. That makes it difficult to determine the channel posts pagination loading state inside of the reducer.

For now, I think we should roll with this and we can move the logic in a refactoring PR later.